extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchAggregatorFactory.java

```diff
@@ -289,10 +289,11 @@ public AggregatorFactory substituteCombiningFactory(AggregatorFactory preAggrega
       return null;
     }
     HllSketchAggregatorFactory that = (HllSketchAggregatorFactory) preAggregated;
-    if (lgK == that.lgK && tgtHllType == that.tgtHllType && stringEncoding == that.stringEncoding && Objects.equals(
-        fieldName,
-        that.fieldName
-    )) {
+    if (lgK <= that.lgK &&
+        tgtHllType.ordinal() <= that.tgtHllType.ordinal() &&
```

Comment:
I don't think the `TgtHllType` needs to be part of the check here, since according to the datasketches docs:

> These three target types are isomorphic representations of the same underlying HLL algorithm. Thus, given the same value of lgConfigK and the same input, all three HLL target types will produce identical estimates and have identical error distributions.

& I think that the aggregator is able to merge two different HLL types together.